### PR TITLE
pam-ssh-add: Properly drop auxiliary groups

### PR DESCRIPTION
**Pull request by martinpitt**
Call initgroups() for the destination user to avoid retaining auxilliary
groups after dropping privileges.

Spotted by rpmlint:

    E: missing-call-to-setgroups-before-setuid /usr/lib64/security/pam_ssh_add.so

### CODE REVIEW

**Comment by martinpitt** [comment]
This breaks test-ssh-add (`pam_ssh_add: couldn't setup credentials: Operation not permitted`) and thus needs some more fine-tuning.

**Comment by martinpitt** [comment]
I pushed a fix. It's a bit ugly as it has to put a special case for the unit test into the code, but I don't think it's too bad.

File: src/pam-ssh-add/pam-ssh-add.c [hunk 1]
```diff
@@ -455,8 +456,9 @@ setup_child (const char **args,
   if (setegid (getgid ()) < 0 || seteuid (getuid ()) < 0)
       error ("failed to restore credentials");
 
-  /* Setup process credentials */
-  if (setgid (pwd->pw_gid) < 0 || setuid (pwd->pw_uid) < 0 ||
+  /* Setup process credentials; if we have root privs, drop any auxiliary groups too */
+  if ((geteuid() == 0 ? initgroups(pwd->pw_name, pwd->pw_gid) < 0 : 0) ||
```
Comment (petervo) [inline src/pam-ssh-add/pam-ssh-add.c]: I think that having the uid 0 doesn't always mean you have permission to change the group. Can we base the check on the intent instead? if getgid() != pwd->pw_gid then we call initgroups? Does that work?

Comment (martinpitt) [reply src/pam-ssh-add/pam-ssh-add.c]: Ah, nice idea! Yes, that is better, I updated the commit. (I used `getegid()` though, I think that's slightly more correct).

Comment (petervo) [reply src/pam-ssh-add/pam-ssh-add.c]: We explicitly setegid with getgid above so by this point they should always be the same. But yeah this seems good.

**Pull request merged**